packages/lib-sourcify/src/Verification/Verification.ts

```diff
@@ -0,0 +1,500 @@
+    }
+
+    if (this.creationMatch !== null || this.runtimeMatch !== null) {
+      logInfo('Verified contract', {
+        address: this.address,
+        chainId: this.sourcifyChain.chainId,
+        runtimeMatch: this.runtimeMatch,
+        creationMatch: this.creationMatch,
+      });
+      return;
+    }
+
+    throw new VerificationError(
+      "The deployed and recompiled bytecode don't match.",
+      'NO_MATCH',
+    );
+  }
+
+  async handleSolidityBugCases(
+    forceEmscripten: boolean,
+    compiledRuntimeBytecode: string,
+  ): Promise<SolidityBugType> {
+    // Handle Solidity specific verification bug cases
+    if (
+      this.compilation instanceof SolidityCompilation &&
+      this.runtimeMatch === null
+    ) {
+      const settings = this.compilation.jsonInput.settings;
+
+      // Handle when <0.8.21 and with viaIR and with optimizer disabled
+      // See issues:
+      //   https://github.com/ethereum/sourcify/issues/1088
+      if (
+        !forceEmscripten && // Enter this case only if we are not already forcing Emscripten
+        lt(this.compilation.compilerVersion, '0.8.21') &&
+        !settings.optimizer?.enabled &&
+        settings.viaIR
+      ) {
+        logInfo('Force Emscripten compiler', {
+          address: this.address,
+          chainId: this.sourcifyChain.chainId,
+        });
+
+        // Try to verify again with Emscripten
+        return SolidityBugType.ECMASCRIPT_BUG;
+      }
+
+      // Case when extra unused files in compiler input cause different bytecode
+      // See issues:
+      //   https://github.com/ethereum/sourcify/issues/618
+      //   https://github.com/ethereum/solidity/issues/14250
+      //   https://github.com/ethereum/solidity/issues/14494
+      try {
+        const [, deployedAuxdata] = splitAuxdata(
+          this.onchainRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        const [, recompiledAuxdata] = splitAuxdata(
+          compiledRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        // Metadata hashes match but bytecodes don't match.
+        if (
+          deployedAuxdata === recompiledAuxdata &&
+          settings.optimizer?.enabled
+        ) {
+          throw new VerificationError(
+            "It seems your contract's metadata hashes match but not the bytecodes. You should add all the files input to the compiler during compilation and remove all others. See the issue for more information: https://github.com/ethereum/sourcify/issues/618",
+            'EXTRA_FILE_INPUT_BUG',
+          );
+        }
+      } catch (e: any) {
+        if (e instanceof VerificationError) {
+          throw e;
+        }
+        logWarn('Error checking for extra-file-input-bug', {
+          chain: this.sourcifyChain.chainId,
+          address: this.address,
+          error: e.message,
+        });
+        throw new VerificationError(
+          e.message,
+          'FAILED_TO_CHECK_EXTRA_FILE_INPUT_BUG',
+        );
+      }
```

Comment:
Why do we have this explicit check? Is there a known case of such errors? Seemed unnecessary to me

Reply:
Yes I think that's a little unnecessary. I don't think it's likely anything will go wrong here. We can propagate the error without catching or we could have a catch-all clause with a generic VerificationError. `FAILED_TO_CHECK_EXTRA_FILE_INPUT_BUG` seemed a bit too much

Reply:
I'll just remove the error for now